support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go

```diff
@@ -161,32 +202,91 @@ func (r *PodReconciler) updateorCreatePodEntry(ctx context.Context, pod *corev1.
 }
 
 // podSpiffeID returns the desired spiffe ID for the pod, or nil if it should be ignored
-func (r *PodReconciler) podSpiffeID(pod *corev1.Pod) string {
+func (r *PodReconciler) podSpiffeID(pod *corev1.Pod) (string, error) {
 	if r.c.PodLabel != "" {
 		// the controller has been configured with a pod label. if the pod
 		// has that label, use the value to construct the pod entry. otherwise
 		// ignore the pod altogether.
 		if labelValue, ok := pod.Labels[r.c.PodLabel]; ok {
-			return makeID(r.c.TrustDomain, "%s", labelValue)
+			return makeID(r.c.TrustDomain, "%s", labelValue), nil
 		}
-		return ""
+		return "", nil
 	}
 
 	if r.c.PodAnnotation != "" {
 		// the controller has been configured with a pod annotation. if the pod
 		// has that annotation, use the value to construct the pod entry. otherwise
 		// ignore the pod altogether.
 		if annotationValue, ok := pod.Annotations[r.c.PodAnnotation]; ok {
-			return makeID(r.c.TrustDomain, "%s", annotationValue)
+			return makeID(r.c.TrustDomain, "%s", annotationValue), nil
+		}
+		return "", nil
+	}
+
+	// the controller has not been configured with a pod label or a pod annotation, so identity_template must be set
+	// as it is enforced in config_crd
+	if r.c.IdentityTemplate == "" {
+		// this is just a sanity check
+		return "", nil
+	}
+
+	// if identity_template_label is not provided, all pods get the template formatted SVID
+	if r.c.IdentityTemplateLabel != "" {
+		// if identity_template_label is provided, only pods that have this label=`true` get the template formatted SVID
+		if labelValue, ok := pod.Labels[r.c.IdentityTemplateLabel]; ok {
+			if !strings.EqualFold("true", labelValue) {
+				return "", nil
+			}
+		} else {
+			return "", nil
 		}
-		return ""
 	}
 
-	// the controller has not been configured with a pod label or a pod annotation.
-	// create an entry based on the service account.
-	return makeID(r.c.TrustDomain, "ns/%s/sa/%s", pod.Namespace, pod.Spec.ServiceAccountName)
+	// create an entry using provided identity template.
+	svid, err := r.getIdentityTemplate(pod)
+	if err != nil {
+		return "", err
+	}
+	return makeID(r.c.TrustDomain, svid), nil
 }
 
 func (r *PodReconciler) podParentID(nodeName string) string {
 	return makeID(r.c.TrustDomain, "k8s-workload-registrar/%s/node/%s", r.c.Cluster, nodeName)
 }
+
+func (r *PodReconciler) getIdentityTemplate(pod *corev1.Pod) (string, error) {
```

Comment:
The name `getIdentityTemplate()` is a bit inaccurate here because the result is the actual SPIFFE ID path generated from the template + pod + context.
```suggestion
func (r *PodReconciler) generateSpiffeIDPath(pod *corev1.Pod) (string, error) {
```